### PR TITLE
fix(shell-api): do not forward auth errors on completion MONGOSH-719

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: packages/cli-repl/src/cli-repl.spec.ts [hunk 1]
```diff
@@ -799,6 +810,13 @@ describe('CliRepl', () => {
         }
       });
 
+      it('completes the db.version method', async() => {
+        output = '';
+        input.write('db.vers\u0009\u0009');
```
Comment (mcasimir) [inline packages/cli-repl/src/cli-repl.spec.ts:815]: Just super random thought .. we could have `const TAB = "\u0009";` and `db.vers${TAB}${TAB}` or something .. not a life changing one :)

File: packages/shell-api/src/shell-internal-state.ts [hunk 2]
```diff
@@ -308,7 +308,7 @@ export default class ShellInternalState {
           const collectionNames = await this.currentDb._getCollectionNamesForCompletion();
           return collectionNames.filter((name) => name.startsWith(collName));
         } catch (err) {
-          if (err.code === ShellApiErrors.NotConnected) {
+          if (err.code === ShellApiErrors.NotConnected || err.codeName === 'Unauthorized') {
```
Comment (mcasimir) [inline packages/shell-api/src/shell-internal-state.ts:311]: Should `'Unauthorized'` be also part `ShellApiErrors` to be consistent with the other?

**Review by mcasimir — APPROVED** [review]

File: packages/shell-api/src/shell-internal-state.ts (hunk 2 above)
Comment (addaleax) [reply packages/shell-api/src/shell-internal-state.ts:311]: This is not a shell error, though, it’s a server error :)

File: packages/cli-repl/src/cli-repl.spec.ts (hunk 1 above)
Comment (addaleax) [reply packages/cli-repl/src/cli-repl.spec.ts:815]: Sounds good, we already do that for e.g. arrow-up in the other tests. I’ll do that right after merging this.

File: packages/shell-api/src/shell-internal-state.ts (hunk 2 above)
Comment (mcasimir) [reply packages/shell-api/src/shell-internal-state.ts:311]: Makes sense!

**Pull request merged**